bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/VersionedClientlibsTransformerFactory.java

```diff
@@ -272,10 +274,26 @@ private String getVersionedPath(final String originalPath, final LibraryType lib
     }
 
     private HtmlLibrary getLibrary(LibraryType libraryType, String libraryPath, ResourceResolver resourceResolver) {
-        String resolvedLibraryPath = resolvePathIfProxied(libraryType, libraryPath, resourceResolver);
+        String resolvedLibraryPath = resolvePath(libraryType, libraryPath, resourceResolver);
         return resolvedLibraryPath == null ? null : htmlLibraryManager.getLibrary(libraryType, resolvedLibraryPath);
     }
 
+    private String resolvePath(LibraryType libraryType, String libraryPath, ResourceResolver resourceResolver) {
+        String clientLibraryPath = resolvePathIfJcrMapping(libraryPath, resourceResolver);
+        if (StringUtils.isEmpty(clientLibraryPath)) {
+            clientLibraryPath = resolvePathIfProxied(libraryType, libraryPath, resourceResolver);
+        }
+        return clientLibraryPath;
+    }
+
+    private String resolvePathIfJcrMapping(String libraryPath, ResourceResolver resourceResolver) {
```

Comment:
The method name is a bit off as you always do the mapping. I would prefer `resolvePath`, also I think the methods `resolvePath` and `resolvePathIfJcrMapping` should be merged into one.

Reply:
Methods merged in one "resolvePath" that will take care of JCR resolve and If Proxied